scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -196,115 +197,135 @@ public int schedule() {
 
             // ask the policy all the tasks to be schedule according to the jobs list.
 
-            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
+            LinkedList<EligibleTaskDescriptor> fullListOfTaskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
 
             //if there is no task to scheduled, return
-            if (taskRetrievedFromPolicy == null || taskRetrievedFromPolicy.isEmpty()) {
+            if (fullListOfTaskRetrievedFromPolicy == null || fullListOfTaskRetrievedFromPolicy.isEmpty()) {
                 return numberOfTaskStarted;
             }
 
-            if (logger.isDebugEnabled()) {
-                logger.debug("eligible tasks : " + taskRetrievedFromPolicy);
-            }
-
             schedulingService.unlockJobsToSchedule(toUnlock.values());
             toUnlock = null;
 
-            updateVariablesForTasksToSchedule(jobMap, taskRetrievedFromPolicy);
+            int subTasksListFrom = 0;
+            int subTasksListTo = Math.min(fullListOfTaskRetrievedFromPolicy.size(),
```

Comment:
You can precise even more the boundaries like `subTasksListFromInclusive` and `subTasksListToExclusive`.

Reply:
Good idea, always better to clarify the boundaries :)
I will remember it 👍